source/blog/_posts/2018-11-30-how-to-use-phive.md

```diff
@@ -0,0 +1,74 @@
+---
+title: What is phive and why you should use it!
+tags:
+- Deployment
+- Phive
+- Phar
+
+categories: 
+- dev
+
+authors: [tn]
+github_link: /blog/_posts/2018-11-30-how-to-use-phive.md
+
+---
+
+At the beginning of this year we decided to create a new enterprise accelerator for our enterprise eco system. The so called Pricing Engine. With this new project we evaluated the PHAR Installation and Verification Environment (PHIVE) to manage all our *.phar dev tools.
+
+Normally we store our dev tools in the document root of our git repository. As an example in the b2b suite we have seven several phar files (deptrac, phpDocumentator, phpcpd, phploc, phpstan, psh and security-checker). With phive we can manage all our dev tools with a phive.xml file. As you can see in the below example, you can define the version which should be used, the position where it should be placed and at last if it should be a symlink or a real copy of the phar file. Normally phive downloads the specific phar file and symlinks it in the place where it has to be.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<phive xmlns="https://phar.io/phive">
+  <phar name="php-cs-fixer" version="^2.12.2" installed="2.13.0" location="./tools/php-cs-fixer" copy="true"/>
+  <phar name="shopwarelabs/psh" version="^1.2.0" installed="1.2.0" location="psh" copy="true"/>
+  <phar name="phpstan" version="^0.10.1" installed="0.10.3" location="./tools/phpstan" copy="true"/>
+  <phar name="shopwarelabs/sw-cli-tools" version="^0.0.1" installed="0.0.1" location="./tools/sw" copy="true"/>
+</phive>
+```  
+
+This sounds very interesting isn't it? But how difficult is it to create a new software project and use phive for managing all dev tools?
+
+## Setup a new Project with phive
+The question is very easy to answer, it is as easy as to create a new composer project. As a preparation we have to install phive on our system. To do it use the snippet below:
+
+```bash
+wget -O phive.phar https://phar.io/releases/phive.phar
+wget -O phive.phar.asc https://phar.io/releases/phive.phar.asc
+gpg --keyserver pool.sks-keyservers.net --recv-keys 0x9D8A98B29B2D5D79
+gpg --verify phive.phar.asc phive.phar
+chmod +x phive.phar
+sudo mv phive.phar /usr/local/bin/phive
+```
+
+After the installation we can easily install php-cs-fixer with ```phive install php-cs-fixer```. In this example php-cs-fixer is an alias and corresponds to the GitHub repository [FriendsOfPHP/PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer). You can find alternative installation instructions below:
+
+```bash
```

Comment:
Can you use the same dependency in the alternatives, just to show how to use the alternatives?

```suggestion
```bash
phive install php-cs-fixer
phive install FriendsOfPHP/PHP-CS-Fixer
phive install https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar
```